components/provisioner/internal/operations/queue/queues_constructor.go

```diff
@@ -161,3 +167,26 @@ func CreateShootUpgradeQueue(
 
 	return NewQueue(upgradeClusterExecutor)
 }
+
+func CreateHibernationQueue(
+	timeouts HibernationTimeouts,
+	factory dbsession.Factory,
+	directorClient director.DirectorClient,
+	shootClient gardener_apis.ShootInterface) OperationQueue {
+
+	waitForHibernation := hibernation.NewWaitForHibernation(shootClient, model.FinishedStage, timeouts.WaitingForClusterHibernation)
```

Comment:
Suggestion: How about changing this name `NewWaitForHibernationStep` for consistency with other queues?